lib/topological_inventory/sync/inventory_upload_validator_worker.rb

```diff
@@ -33,14 +31,42 @@ def perform(message)
       private
 
       def valid_payload?(payload)
-        schema_name = payload.dig("metadata", "schema", "name")
-        schema_klass = schema_klass_name(schema_name).safe_constantize
+        require "json/stream"
+
+        parser = JSON::Stream::Parser.new do
+          key   { |k| puts "key: #{k}" }
+          value { |v| puts "value: #{value}" }
+        end
 
-        schema_klass.present?
+        stream_payload(payload["url"]) do |chunk|
+          parser << chunk
+        end
+
+        # TODO validate that this is a correct topological inventory payload
+        true
       end
 
-      def schema_klass_name(name)
-        "TopologicalInventory::Schema::#{name}"
+      def open_url(url)
+        require "http"
+
+        uri = URI(url)
+        uri.scheme == "file" ? open(uri.path, "rb") : HTTP.get(uri).tap { |resp| resp.body.stream!; resp.body }
```

Comment:
This keeps an open file handle on the File, which we probably want to avoid.  I wonder if we can leverage a block style here using yield to get the autoclosing...I thinking something like...

```ruby
def open_url(url)
  require "http"

  uri = URI(url)
  if uri.scheme == "file" do
    File.open(uri.path, "rb") { |f| yield f }
  else
    yield HTTP.get(uri).tap { |resp| resp.body.stream!; resp.body }
  end
end
```

Then below

```ruby
def stream_payload(url)
  require "rubygems/package"
  open_url(url) do |io|
    Zlib::GzipReader.wrap(io) do |gz|
    ...
  end
end
```

Reply:
Also,

- I don't love the multi-line with the `;`...we can probably flatten that if we drop the ternary.
- We are doing a "rb" here, but what is the encoding when we read over http...if the http one is not binary this can create an inconsistency with testing using a file.
- What does the `resp.body` do?  It's in a tap block, so it must have some side effect?